src/goto-programs/goto_check.cpp

```diff
@@ -248,8 +248,8 @@ void goto_checkt::overflow_check(
 
   // add overflow subgoal
   expr2tc overflow = is_neg2t(expr)
-                       ? expr2tc(new overflow_neg2t(to_neg2t(expr).value))
-                       : expr2tc(new overflow2t(expr));
+                       ? expr2tc(overflow_neg2tc(to_neg2t(expr).value))
+                       : expr2tc(overflow2tc(expr));
```

Comment:
We don't need the expr2tc calls around overflow2tc and overflow_neg2tc, right?

Reply:
Unfortunately we do because otherwise C++ complains about both sides not having equal types. I don't indend to put a lot more effort into something2tc (like conversion operators or the like), because in a later PR I would like to remove it.

Reply:
This could actually go away if we remove something2tc and make the *2tc() things function calls.